src/main/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticator.java

```diff
@@ -104,11 +107,11 @@ public HTTPSamlAuthenticator(final Settings settings, final Path configPath) {
                 throw new Exception("kibana_url is unconfigured");
             }
 
-            if (idpMetadataUrl == null && idpMetadataFile == null) {
-                throw new Exception("idp.metadata_url and idp.metadata_file are unconfigured");
+            if (idpMetadataUrl == null && idpMetadataFile == null && idpMetadataBody == null) {
+                throw new Exception("idp.metadata_url, idp.metadata_file and idpMetadataBody are unconfigured");
```

Comment:
Consider "One of idp.metadata_url, idp.metadata_file or idp.metadata_body must be configured".